framework/src/dm/arch/sidk_s5jt200/s5j_dm_connectivity.c

```diff
@@ -385,3 +397,201 @@ int dm_conn_unregister_linkdown_cb(conn_cb cb)
 	dmdbg("Link Down Callback UnRegistered Successfully\n");
 	return DM_ERROR_NONE;
 }
+
+static int8_t prv_dm_conn_scan_result(slsi_reason_t *reason)
+{
+	if (reason->reason_code != 0) {
+		dmdbg("Failed to scan : reason %d, locally generated : %d\n", \
+			  reason->reason_code, reason->locally_generated);
+		wifiscan_result = NULL;
+	} else {
+		WiFiFreeScanResults(&wifiscan_result);
+		WiFiGetScanResults(&wifiscan_result);
+		WiFiNetworkJoin((uint8_t *) CONFIG_DM_AP_SSID, strlen(CONFIG_DM_AP_SSID), \
+						NULL, (slsi_security_config_t *) getSecurityConfig(CONFIG_DM_AP_SECURITY, \
+								CONFIG_DM_AP_PASS, SLSI_WIFI_STATION_IF));
+	}
+	printf("scanned result\n");
+	return SLSI_STATUS_SUCCESS;
+}
+
+int dm_conn_get_scan_result(dm_scan_info_t **result)
+{
+	dm_scan_info_t *curr_record;
+	dm_scan_info_t *prev_record = NULL;
+	if (wifiscan_result == NULL) {
+		return DM_ERROR_NO_DATA;
+	}
+	/* Initialize pointer */
+	iterator = wifiscan_result;
+
+	do {
+		curr_record = (dm_scan_info_t *)malloc(sizeof(dm_scan_info_t));
+		if (curr_record == NULL) {
+			dm_conn_free_scan_result(result);
+			goto error_with_record;
+		}
+		if (!prev_record) {
+			*result = curr_record;
+		} else {
+			prev_record->next = curr_record;
+		}
+
+		memset(curr_record->ssid, 0x00, SLSI_SSID_LEN + 1);
+		memset(curr_record->bssid, 0x00, DM_BSSID_LEN);
+		curr_record->rssi = iterator->rssi;
+		strncpy(curr_record->ssid, (char *) iterator->ssid, strlen((const  char*)iterator->ssid));
+		strncpy(curr_record->bssid, (char *)iterator->bssid, strlen((const  char*)iterator->bssid));
+		prev_record = curr_record;
+		iterator = iterator->next;
```

Comment:
Line 430 checks for record.
line 422 checks for initial value of iterator. If you are referring to iterator->ssid and bssid, then they come from wifi, and we assume that part is checked already at wifi supplicant.